### PR TITLE
Polynomial Multiproof RPC

### PR DESCRIPTION
**Pull request by Aphoh**
This PR pulls in a big refactor in `avail-core`, and makes some big changes in the RPC implementation:
1. Switches from `Mutex<LRU>` to an asynchronous, performant cache backing grid generation in the api, which [coalesces requests for the same grid](https://docs.rs/moka/latest/moka/sync/struct.Cache.html#concurrent-calls-on-the-same-key-2). This means the DA grid is only constructed once, even with many parallel requests inbound.
2. Caches the ifft-d rows of the grid used in proof generation.
3. Allows easily for the caching proofs, but not currently implemented 😄 

Additionally, it adds one new rpc endpoint, `kate_queryMultiProof` which spits out multiproofs.

### CODE REVIEW

**Comment by Aphoh** [comment]
Added multiproof retrieval and verification to the data submission example [here](https://github.com/availproject/avail/blob/149aaf1989d84a8a0ac42a69e4d903e7bd5b958b/avail-subxt/examples/submit_data.rs#L135)
@kroos47 Let me know if you see any blockers/anything you'd like for getting this worked into the light client.

File: Cargo.toml [hunk 1]
```diff
@@ -22,15 +23,15 @@ frame-system-rpc-runtime-api = { path = "pallets/system/rpc/runtime-api" }
 frame-system-benchmarking = { path = "pallets/system/benchmarking" }
 
 # DA Primitives
-da-primitives = { version = "0.4", git="https://github.com/availproject/avail-core", tag = "da-primitives/v0.4.6" }
-kate = { version = "0.7", git="https://github.com/availproject/avail-core", tag = "da-primitives/v0.4.6" }
-kate-recovery = { version = "0.8", git="https://github.com/availproject/avail-core", tag = "da-primitives/v0.4.6" }
+da-primitives = { version = "0.4", git="https://github.com/maticnetwork/avail-core",  rev = "5bd30863f15cca7b0158319979202650dc775d04" }
```
Comment (fmiguelgarcia) [inline Cargo.toml:26]: We need to fix this deps.
We have to use `availproject` and specific `tag` instead of `rev`

**Review by fmiguelgarcia — CHANGES_REQUESTED** [review]

File: pallets/system/Cargo.toml [hunk 2]
```diff
@@ -49,7 +50,9 @@ default = ["std"]
 std = [
 	"avail-base/std",
 	"kate/std",
+  "kate/parallel",
```
Comment (fmiguelgarcia) [inline pallets/system/Cargo.toml:53]: nit: spaces/tabs

File: pallets/system/src/header_builder.rs [hunk 3]
```diff
@@ -126,23 +122,45 @@ pub fn build_extension<M: Metrics>(
...
+	static PMP: Lazy<kate::pmp::m1_blst::M1NoPrecomp> =
+		once_cell::sync::Lazy::new(|| kate::testnet::multiproof_params(256, 256));
 	use da_primitives::header::extension::{v1, v2};
 
-	let (xts_layout, commitment, block_dims, _data_matrix) = kate::com::par_build_commitments(
-		block_length.rows,
-		block_length.cols,
-		block_length.chunk_size(),
-		app_extrinsics,
+	let grid = kate::gridgen::EvaluationGrid::from_extrinsics(
```
Comment (fmiguelgarcia) [inline pallets/system/src/header_builder.rs:130]: @Aphoh We need to merge your changes in `avail-core`, tag them and finally use here as dep.

**Review by fmiguelgarcia — CHANGES_REQUESTED** [review]

**Comment by markopoloparadox** [comment]
@Aphoh Should we rebase this branch? Is this PR tied to the to other PR that you have created https://github.com/availproject/avail/pull/196 ? :)

**Comment by Aphoh** [comment]
Closing in favor of #254

**Pull request closed without merging**